en/docs/install-and-setup/setup/deployment-best-practices/performance-and-capacity-planning.md

```diff
@@ -0,0 +1,260 @@
+  </tr>
+  <tr>
+   <td>Average Memory Footprint After Full GC (M)
+   </td>
+   <td>The average memory consumed by the application after a full garbage collection event.
+   </td>
+  </tr>
+</table>
+
+For a detailed analysis of the performance of API-M 3.2.0, see [API-M 3.2.0 Performance graphs on Github](https://github.com/wso2/performance-apim/tree/performance-test-276-2020-08-03_08-47-25/performance/benchmarks/3.2.0%20graphs-all).
+
+## Observations from all results
+
+There are key observations for the average user scenario of accessing APIs with 1KiB messages and the back-end service having 30ms delay.
+
+The following are the key observations from the all performance tests done with different message sizes and different backend delays. (See **Comparison of results** for all charts used to derive the pointed mentioned below)
+
+#### Throughput comparison:
+
+A throughput increase is observed in the transformation scenario in API-M 3.2.0, in comparison to API-M 3.1.0
+
+The throughput increases up to a certain limit when the number of concurrent users increases. The Mediation API throughput increase rate is much lower than the Echo API. Throughput decreases when the message sizes increase. Throughput decreases when the backend sleep time increase. This observation is similar to both APIs. This means that if the backend takes more time, the request processing rate at the API Manager Gateway will be less.
+
+#### Key observations related to response time:
+
+The average response time increases when the number of concurrent users increases. The increasing rate of average response time for both API-M 3.2.0 and API-M 3.1.0 is similar.
+
+The average response time increases when the number of concurrent users increases. The average response time increases considerably for Mediation API when the message sizes increase due to the message processing. The average response time of the Echo API does not increase as much as the Mediation API. The average response time increases when the backend sleep time increases. This observation is similar to both APIs.
+
+#### Key observations related to GC throughput:
+
+The GC throughput decreases when the number of concurrent users increases. When there are more concurrent users, the object allocation rate increases. The GC throughput increases when the message size increases. The request processing rate slows down due to the time taken to process large messages. Therefore, the object allocation rate decreases  when the message size increases. The GC throughput increases when the backend sleep time increases. The object allocation rate will be low when the backend takes more time to respond.
+
+## Comparison of 3.1.0 and 3.2.0
+
+### Average response time comparison
+
+#### Average response time vs concurrent users
+
+  [![]({{base_path}}/assets/img/setup-and-install/performance-test-results/average_time_0ms_50B.png)]({{base_path}}/assets/img/setup-and-install/performance-test-results/average_time_0ms_50B.png)
```

Comment:
Best to add values within the [   ] tags which will serve as ALT for the images.